### PR TITLE
Add sample estimation

### PR DESCRIPTION
**Pull request by andrewemark**
Adds sample estimation and tests.

Note that the randomization option on `submit` is removed such that performing estimation followed by submit will occur with consistent job parameters. Now, unspecified params obtain their default values, and randomization but be added manually with a function call.

Also note the TODO for updating documentation. We have no public generators with sample estimation functionality, so it doesn't really make sense to document it in this repo yet. We will document in the workflows repo, however.

Lastly, pardon the whitespace changes, but I ran all the automated tools required by the build -- isort, mypy, black.

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -179,3 +207,31 @@ def test_standard_batch(self, cfg: TestConfig) -> None:
...
+            "Test estimation",
+            cfg.server,
+        )
+        assert estimate == [1, 1]
+
+    def test_estimate_standard_batch(self, cfg: TestConfig) -> None:
+        estimate = estimate_batch_samples(
+            cfg.token,
+            cfg.generator_name,
+            JobType.PREVIEW,
```
Comment (vnerozinea) [inline tests/test_integration.py:230]: What is the difference between "test_estimate_standard_batch" and "test_estimate_preview_batch"?

File: tests/test_integration.py [hunk 2]
```diff
@@ -151,6 +151,34 @@ def test_post_multi_job_batch(self, cfg: TestConfig) -> None:
...
+@pytest.mark.integration
+@pytest.mark.apipost
+class TestApiSampleEstimateRequestIntegration:
+    def test_estimate_batch_preview_request(self, cfg: TestConfig) -> None:
+        r = api.estimate_batch_samples(
+            token=cfg.token,
+            generator=cfg.generator_name,
+            name="test",
+            job_params=[{}],
+            is_preview=False,
```
Comment (vnerozinea) [inline tests/test_integration.py:163]: "test_estimate_batch_preview_request" but is_preview=False

File: tests/test_integration.py (hunk 1 above)
Comment (andrewemark) [reply tests/test_integration.py]: I had a typo in the call to `estimate_batch_samples`. Fixed that. One test is for previews, the other is for videos.

File: tests/test_integration.py (hunk 2 above)
Comment (andrewemark) [reply tests/test_integration.py]: Fixed. Good catch

**Review by vnerozinea — APPROVED** [review]

**Pull request merged**